Document/0x05b-Basic-Security_Testing.md

```diff
@@ -1244,13 +1244,43 @@ For enterprise tools, see the section "Static Source Code Analysis" in the chapt
 
 #### Dynamic Analysis
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1240>
+Dynamic Analysis tests the release version of an application, in a similar fashion to black box application testing. The way the code works should be inferred and looked at from a higher level, unlike static analysis, which takes the application's code and tries to analyze how the application will behave. This allows the tester to properly identify:
+
+- Business logic flaws which extend over multiple services
+- Vulnerabilities in live environments
+- Weak input validation and bad input/output encoding as they are processed through one or multiple services
+
+ whereas static analysis would identify:
+
+- Code bad practices
+- Regex usage issues
+- Hidden code layers
+
+Analysis can be assisted by automated tools, such as [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF/), while assessing an application. An application can be assessed by side-loading it, re-packaging it, or by simply attacking the installed version.
 
 ##### Using Non-Rooted Devices
 
-##### Method Tracing
+Non-rooted devices provide the tester with two benefits:
+
+- Replicate the same environment of the installed application
+- Use any device without the need of having it rooted
```

Comment:
It is a benefit for a tester of not having it rooted, therefore please rewrite:
- As a tester you don't need to bypass root detection and can use objection to package the app with the Frida Gadget.